frontend/src/component/common/Chart/customHighlightPlugin.ts

```diff
@@ -12,23 +38,27 @@ export const customHighlightPlugin = (bottomOverflow = 34) => ({
                 x,
                 yAxis.top,
                 x,
-                yAxis.bottom + 34,
+                yAxis.bottom + bottomOverflow,
             );
             gradient.addColorStop(0, 'rgba(129, 122, 254, 0)');
-            gradient.addColorStop(1, 'rgba(129, 122, 254, 0.12)');
+            gradient.addColorStop(
+                1,
+                `rgba(129, 122, 254, ${maxHighlightOpacity})`,
+            );
             ctx.fillStyle = gradient;
 
-            const barWidth =
-                (chart.width / (chart.data.labels?.length ?? 1)) *
-                (chart.options.datasets?.bar?.categoryPercentage ?? 1);
-            ctx.roundRect(
-                x - barWidth / 2,
+            const args: [number, number, number, number] = [
+                x - highlightWidth / 2,
                 yAxis.top,
-                barWidth,
+                highlightWidth,
                 yAxis.bottom - yAxis.top + bottomOverflow,
-                5,
-            );
-            ctx.fill();
+            ];
+            if (bottomOverflow) {
+                ctx.roundRect(...args, 5);
+                ctx.fill();
+            } else {
+                ctx.fillRect(...args);
```

Comment:
These cases are slightly different. If there is overflow, draw a rounded rectangle. If not, draw a perfect rectangle. Also, the drawing functions have slightly different behavior, which causes a weird full overlay at times with roundRect on the analytics page.